### PR TITLE
Delete objects

### PR DESCRIPTION
**Pull request by louen**
This PR has two commits : one is a minor fix to Cmake messages, and the other implements correct object deletion order (#153) and a clear scene button and method(#154)

### CODE REVIEW

File: src/Core/Index/IndexMap.hpp [hunk 1]
```diff
@@ -45,6 +45,11 @@ class IndexMap {
     inline Index insert( const T& obj );                    // Insert a object in the IndexMap. Return an invalid index if the object is not inserted.
     inline bool  insert( const T& obj, Index& idx );        // Insert a object in the IndexMap. Return true if the object is inserted, and store it in idx.
 
+    template<typename... Args>
+    inline Index emplace( const Args&&... args );             // Construct an object in place in the IndexMap. Return an invalid index if the object is not inserted.
+    template<typename... Args>
+    inline bool  emplace(  Index& idx, const Args&&... args ); // Construct an object in place in the IndexMap. Return true if the object is inserted, and store it in idx.
+
```
Comment (louen) [inline src/Core/Index/IndexMap.hpp]: This was required to use the map with `std::unique_ptr`

File: src/Core/Index/IndexMap.inl [hunk 2]
```diff
@@ -116,7 +145,7 @@ inline bool IndexMap< T >::remove( const uint i ) {
 /// ACCESS
 /// ===============================================================================
 template < typename T >
-inline T IndexMap< T >::at( const Index& idx ) const {
+inline const T& IndexMap< T >::at( const Index& idx ) const {
     typename std::deque< Index >::const_iterator it = std::find( m_index.begin(), m_index.end(), idx );
     CORE_ASSERT( it != m_index.end(), "Index not found" );
     return m_data.at( it - m_index.begin() );
```
Comment (nmellado) [inline src/Core/Index/IndexMap.inl]: Would be cleaner to use std::distance here, no ?
http://en.cppreference.com/w/cpp/iterator/distance

File: src/Core/Index/IndexMap.hpp (hunk 1 above)
Comment (nmellado) [inline src/Core/Index/IndexMap.hpp]: ok.

File: src/Core/Index/IndexMap.inl [hunk 3]
```diff
@@ -301,6 +330,17 @@ inline T& IndexMap< T >::operator[]( const uint i ) {
...
+inline const T& IndexMap< T >::operator[]( const Index& idx ) const {
+    return at( idx );
+}
+
+
+
+template < typename T >
+inline const T& IndexMap< T >::operator[]( const uint i ) const{
+    return at( i );
+}
```
Comment (nmellado) [inline src/Core/Index/IndexMap.inl]: Why do we need a const uint version of this function ?
I sounds better to me to stick to Index, by adding an implicit constructor or cast operator.
By this mean, we know that Index will always be used, without any overhead, which is better for testing and monitoring.

File: src/Engine/Managers/EntityManager/EntityManager.cpp [hunk 4]
```diff
@@ -128,5 +113,18 @@ namespace Ra
                 m_entities[i]->swapTransformBuffers();
             }
         }
+
+        void EntityManager::deleteEntities()
+        {
+            std::vector<uint> indices;
```
Comment (nmellado) [inline src/Engine/Managers/EntityManager/EntityManager.cpp]: suggestion, add:
`indices.reserve(m_entitites.size();)`

File: src/Engine/RadiumEngine.hpp [hunk 5]
```diff
@@ -10,6 +10,7 @@
 #include <vector>
 
 #include <Core/Log/Log.hpp>
+#include <Core/Utils/Singleton.hpp>
```
Comment (nmellado) [inline src/Engine/RadiumEngine.hpp]: Why is this needed ?

Comment (louen) [inline src/Engine/RadiumEngine.hpp]: Actually it's not needed.

**Comment by louen** [comment]
As for the  comments to the Index class I wholeheartedly agree with them, but it would be more sensible to plan for its refactoring in a separate issue.
